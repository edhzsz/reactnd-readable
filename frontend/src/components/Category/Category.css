.category {
  height: 30px;
  width: 100%;
  text-align: center;
  border: 1px solid #AFAFAF;
  cursor: pointer;
}

.category.selected {
  background-color: #baddf9;
}

.category:hover {
  background-color: #baddf9;
}