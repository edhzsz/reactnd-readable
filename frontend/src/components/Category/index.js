import React from 'react';
import PropTypes from 'prop-types';
import './Category.css'

const Category = (props) => {
  return (
    <div className="category">
      {props.name}
    </div>
  );
}

Category.propTypes = {
  name: PropTypes.string.isRequired
}

export default Category;