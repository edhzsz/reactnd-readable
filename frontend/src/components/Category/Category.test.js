import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Category from '.';

const wrapper = shallow(<Category name="React"/>);

describe('Component: Category', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Category name="React" />, div);
  });

  it('should match its snapshot with required props', () => {
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });
});