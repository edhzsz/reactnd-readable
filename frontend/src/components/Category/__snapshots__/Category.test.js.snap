// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Category should match its snapshot with required props 1`] = `
<div
  className="category"
>
  React
</div>
`;
